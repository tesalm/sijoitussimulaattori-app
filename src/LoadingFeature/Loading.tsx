import React from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';

import { login } from '../Auth/actions';
import { User } from '../models';
import { RootState } from '../redux/reducers';
import LoggedInView from './components/HelloUser';
import LoadingView from './components/Loading';
import { loadingStyles as styles } from './styles';

export interface LoadingProps {
  user?: User;
  loginRequest: typeof login;
}

export interface LoadingState {}

export class Loading extends React.Component<LoadingProps, LoadingState> {
  constructor(props: LoadingProps) {
    super(props);
  }

  componentDidMount() {
    this.props.loginRequest();
  }

  render() {
    const { user } = this.props;
    if (user) {
      return (
        <View style={styles.container}>
          <LoggedInView user={user} />
        </View>
      );
    } else {
      return (
        <View style={styles.container}>
          <LoadingView />
        </View>
      );
    }
  }
}

const mapStateToProps = (state: RootState) => ({
  user: state.login.user,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      loginRequest: login,
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Loading);
