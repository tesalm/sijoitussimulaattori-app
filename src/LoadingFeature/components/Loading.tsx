import React from 'react';
import { Text, StyleSheet, TextStyle } from 'react-native';

const LoadingView = (): JSX.Element => {
    return (
      <>
      <Text style={styles.counterTitle}>Loading...</Text>
      <Text style={styles.counterTitle}>(simulated 3 seconds before sending login request)</Text>
      </>
    );
};
  
const styles = StyleSheet.create({
    counterTitle: {
      fontSize: 20,
      textAlign: 'center',
      margin: 1,
    } as TextStyle,
    counterValue: {
      fontSize: 40,
      textAlign: 'center',
      margin: 10,
    } as TextStyle,
  });
  
  export default LoadingView;