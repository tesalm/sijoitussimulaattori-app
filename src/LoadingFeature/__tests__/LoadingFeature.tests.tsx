import * as React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import LoadingView from '../components/Loading';
import { LoadingProps } from '../Loading';

describe('sumUp tests', () => {
  const navigationMock: any = {};
  const middlewares: any = [];
  const mockStore = configureStore(middlewares);

  const loadingProps: LoadingProps = {
    user: undefined,
    loginRequest: jest.fn(),
  };

  it('Loading renders correctly', async () => {
    const component = renderer
      .create(<LoadingView {...loadingProps} {...navigationMock} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
