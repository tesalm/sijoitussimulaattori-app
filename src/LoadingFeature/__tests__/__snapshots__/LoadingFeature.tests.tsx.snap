// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sumUp tests Loading renders correctly 1`] = `
Array [
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 1,
        "textAlign": "center",
      }
    }
  >
    Loading...
  </Text>,
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 1,
        "textAlign": "center",
      }
    }
  >
    (simulated 3 seconds before sending login request)
  </Text>,
]
`;
